pkg/controller/database/postgresqlserver/managed.go

```diff
@@ -154,7 +174,13 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 
 	cr.SetConditions(xpv1.Creating())
 
-	pw, err := e.newPasswordFn()
+	pw, err := e.getPassword(ctx, cr)
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	if pw == "" {
+		pw, err = e.newPasswordFn()
+	}
 	if err != nil {
 		return managed.ExternalCreation{}, errors.Wrap(err, errGenPassword)
 	}
```

Comment:
```suggestion
	if pw == "" {
		pw, err = e.newPasswordFn()
		if err != nil {
			return managed.ExternalCreation{}, errors.Wrap(err, errGenPassword)
		}
	}
	
```

Reply:
I was worrying that reviewers would not like the nested ifs :)
Done.